config/initializers/devise.rb

```diff
@@ -309,3 +310,23 @@
   # changed. Defaults to true, so a user is signed in automatically after changing a password.
   # config.sign_in_after_change_password = true
 end
+
+module Devise
+  module Strategies
+    class JWT < Base
+      def valid? 
+        request.headers["Authorization"].present?
+      end
+
+      def authenticate! 
+        token = request.headers.fetch("Authorization", "").split(" ").last
+        payload = JsonWebToken.decode(token)
+        success! User.find(payload["sub"])
+      rescue ::JWT::ExpiredSignature
+        fail! "Auth token has expired"
+      rescue ::JWT::DecodeError
+        fail! "Auth token is invalid"
+      end
+    end
+  end
+end
```

Comment:
Hi @ShinobiWarior, when I ran it on my local machine, it shown an error messaging saying this devise.rb file missing another `end` closing keyword, so in total there should be 5 `end`